data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/tekuv1/beacon/GetEth1DataCache.java

```diff
@@ -0,0 +1,101 @@
+import io.javalin.plugin.openapi.annotations.HttpMethod;
+import io.javalin.plugin.openapi.annotations.OpenApi;
+import io.javalin.plugin.openapi.annotations.OpenApiContent;
+import io.javalin.plugin.openapi.annotations.OpenApiResponse;
+import java.util.Collection;
+import java.util.List;
+import java.util.function.Function;
+import org.jetbrains.annotations.NotNull;
+import tech.pegasys.teku.api.response.v1.teku.GetEth1DataCacheResponse;
+import tech.pegasys.teku.beaconrestapi.MigratingEndpointAdapter;
+import tech.pegasys.teku.infrastructure.http.HttpErrorResponse;
+import tech.pegasys.teku.infrastructure.json.types.SerializableTypeDefinition;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.EndpointMetadata;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.RestApiRequest;
+import tech.pegasys.teku.spec.datastructures.blocks.Eth1Data;
+import tech.pegasys.teku.validator.coordinator.Eth1DataProvider;
+
+public class GetEth1DataCache extends MigratingEndpointAdapter {
+
+  public static final String ROUTE = "/teku/v1/beacon/pool/eth1cache";
+
+  private static final SerializableTypeDefinition<List<Eth1Data>> ETH1DATA_CACHE_RESPONSE_TYPE =
+      SerializableTypeDefinition.<List<Eth1Data>>object()
+          .name("GetEth1DataCacheResponse")
+          .withField(
+              "data",
+              SerializableTypeDefinition.listOf(Eth1Data.SSZ_SCHEMA.getJsonTypeDefinition()),
+              Function.identity())
+          .build();
+
+  private final Eth1DataProvider eth1DataProvider;
+
+  public GetEth1DataCache(Eth1DataProvider eth1DataProvider) {
+    super(
+        EndpointMetadata.get(ROUTE)
+            .operationId("getTekuV1BeaconPoolEth1cache")
+            .summary("Get all cached Eth1 blocks")
+            .description("Get all cached Eth1 blocks.")
+            .tags(TAG_TEKU)
+            .response(SC_OK, "OK", ETH1DATA_CACHE_RESPONSE_TYPE)
+            .withNotFoundResponse()
+            .build());
+    this.eth1DataProvider = eth1DataProvider;
+  }
+
+  @OpenApi(
+      path = ROUTE,
+      method = HttpMethod.GET,
+      summary = "Get all cached Eth1 blocks",
+      description = "Get all cached Eth1 blocks.",
+      tags = {TAG_TEKU},
+      responses = {
+        @OpenApiResponse(
+            status = RES_OK,
+            content = @OpenApiContent(from = GetEth1DataCacheResponse.class)),
+        @OpenApiResponse(status = RES_NOT_FOUND),
+        @OpenApiResponse(status = RES_INTERNAL_ERROR)
+      })
+  @Override
+  public void handle(@NotNull Context ctx) throws Exception {
+    adapt(ctx);
+  }
+
+  @Override
+  public void handleRequest(RestApiRequest request) throws JsonProcessingException {
+    request.header(Header.CACHE_CONTROL, CACHE_NONE);
+    Collection<Eth1Data> eth1CachedBlocks = this.eth1DataProvider.getEth1CachedBlocks();
+    if (eth1CachedBlocks.isEmpty()) {
+      request.respondWithCode(
+          SC_NOT_FOUND, new HttpErrorResponse(SC_NOT_FOUND, "Eth1 blocks cache is empty"));
```

Comment:
commented in the new function, but basically this is just `request.respondError(SC_NOT_FOUND, "Eth1 block cache is empty")`